R/automation.R

```diff
@@ -128,45 +132,108 @@ setMethod("scriptSavepoint", "Script", function(x) {
 #' # But more details are available with function:
 #' showScriptErrors()
 #'
-#' # After a successful run, can look at scripts
+#' # After a successful run, can look at scripts:
 #' scripts(ds)
+#' 
+#' # Run Crunch Automation on a folder:
+#' my_folder <- cd(projects(), "folder1")
+#' runCrunchAutomation(my_folder, 'CREATE FOLDER "folder2";')
 #'
 #' }
-#' @export
 #' @seealso [`automation-undo`] & [`script-catalog`]
+#' @export
 runCrunchAutomation <- function(
-    dataset,
+    x,
     script,
     is_file = string_is_file_like(script),
     encoding = "UTF-8",
     ...
-) {
-    reset_automation_error_env()
-    stopifnot(is.dataset(dataset))
-    stopifnot(is.character(script))
-
-    if (is_file) {
-        automation_error_env$file <- script
-        script <- readLines(script, encoding = encoding, warn = FALSE)
-    } else {
-        automation_error_env$file <- NULL
+){
+    dots <- list(...)
+
+    # a previous version of this function had `dataset` as its first argument
+    # because the function is now broader, the argument is called `x`
+    # the following block is for backwards compatibility
+    args      <- as.list(sys.call())
+    args[[1]] <- NULL
+    if ('dataset' %in% names(args)){
+        warning(
+            'The argument `dataset` has been renamed to `x` because the function ',
+            'now supports more generic operations. ',
+            'While using `dataset` will continue to work, it will emit this warning.'
+        )
+        target <- list(x = args$dataset)
+        args$dataset <- NULL
+        dots$dataset <- NULL
+        do.call(
+            'runCrunchAutomation', c(target, args, dots),
+            quote = FALSE, envir = parent.frame()
+        )
+        return(invisible(NULL))
```

Comment:
why not return result of the ultimate `runCrunchAutomation()`?

Reply:
OK